sklearn/utils/_param_validation.py

```diff
@@ -415,31 +421,137 @@ def __str__(self):
         )
 
 
-def generate_invalid_param_val(constraint):
+def generate_invalid_param_val(constraint, constraints=None):
     """Return a value that does not satisfy the constraint.
 
+    Raises a NotImplementedError if there exists no invalid value for this constraint.
+
     This is only useful for testing purpose.
 
     Parameters
     ----------
-    constraint : Constraint
+    constraint : _Constraint instance
         The constraint to generate a value for.
 
+    constraints : list of _Constraint instances or None, default=None
+        The list of all constraints for this parameter. If None, the list containt only
+        the constraint is used.
```

Comment:
```suggestion
    constraints : list of _Constraint instances or None, default=None
        The list of all constraints for this parameter. If None, the list only
        contains the constraints which is used.
```

Reply:
My sentence had a typo but was not clear at all. Here's what I meant. Is it clearer ?
```suggestion
    constraints : list of _Constraint instances or None, default=None
        The list of all constraints for this parameter. If None, the list only containing
        `constraint` is used.
```